firehose-template.yaml

```diff
@@ -157,6 +157,14 @@ Resources:
                 Action:
                   - logs:PutLogEvents
                 Resource: 'arn:aws:logs:*:*:*'
+        - PolicyName: Firehose-SecretsManagerAccess
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 'secretsmanager:GetSecretValue'
+                Resource: !If [ShouldCreateSecret, !Ref NewRelicLogsLicenseKeySecret, ['*'] ]
```

Comment:
As discussed offline, we should not create this policy when we are not using secrets:
try something like this
```
- !If 
  - ShouldCreateSecret
  - PolicyName: Firehose-SecretsManagerAccess
    PolicyDocument:
      Version: 2012-10-17
      Statement:
        - Effect: Allow
          Action:
            - 'secretsmanager:GetSecretValue'
          Resource: !Ref NewRelicLogsLicenseKeySecret
  - !Ref "AWS::NoValue"

Reply:
Tried that, and it's working. Implemented in code.